experimental/listener.go

```diff
@@ -6,6 +6,24 @@ import (
 	"github.com/tetratelabs/wazero/api"
 )
 
+// StackIterator allows iterating on each function of the call stack, starting
+// from the top. At least one call to Next() is required to start the iteration.
+//
+// Example:
+//
+//	for it.Next() {
+//		fmt.Printf("function: %s, args: %v", it.FnType(), it.Args())
+//	}
+type StackIterator interface {
+	// Next moves the iterator to the next function in the stack. Returns false
+	// if it reached the bottom of the stack.
+	Next() bool
+	// FnType returns the function type of the current function.
```

Comment:
rename to FunctionDefinition and then you can comment like this. We typically don't abbreviate Function in public symbols.

```suggestion
	// FunctionDefinition returns the same value as the same method on api.Function
```

Reply:
👍🏻 Renamed